internal/scheduler/scheduler.go

```diff
@@ -774,9 +776,8 @@ func (s *Scheduler) generateUpdateMessagesFromJob(job *jobdb.Job, jobRunErrors m
 		events = append(events, jobReprioritised)
 	}
 
-	if origJob != job {
-		err := txn.Upsert([]*jobdb.Job{job})
-		if err != nil {
+	if !origJob.Equal(job) {
```

Comment:
The previous behaviour would be to always upsert jobs we've made any changes to even if those changes were no-ops.